datasets/wit/wit.py

```diff
@@ -0,0 +1,99 @@
+import gzip
+import json
+
+import datasets
+from datasets import Features, Sequence, Value
+
+
+_CITATION = """@article{srinivasan2021wit,
+  title={WIT: Wikipedia-based Image Text Dataset for Multimodal Multilingual Machine Learning},
+  author={Srinivasan, Krishna and Raman, Karthik and Chen, Jiecao and Bendersky, Michael and Najork, Marc},
+  journal={arXiv preprint arXiv:2103.01913},
+  year={2021}
+}
+"""
+_DESCRIPTION = """Wikipedia-based Image Text (WIT) Dataset is a large multimodal multilingual dataset. WIT is composed of a curated set
+ of 37.6 million entity rich image-text examples with 11.5 million unique images across 108 Wikipedia languages. Its
+ size enables WIT to be used as a pretraining dataset for multimodal machine learning models.
+"""
+_HOMEPAGE = "https://github.com/google-research-datasets/wit"
+_URL = "https://storage.googleapis.com/huggingface-nlp/datasets/wit/"
+_URLS = {"train": [_URL + f"part-{'%05d' % i}-48a6f07e-bb86-4735-aac7-883349f41a28-c000.json.gz" for i in range(400)]}
+
+
+class Wit(datasets.GeneratorBasedBuilder):
+    """WIT: Wikipedia-based Image Text Dataset for Multimodal Multilingual Machine Learning"""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=Features(
+                {
+                    "b64_bytes": Value("string"),
+                    "embedding": Sequence(Value("float64")),
+                    "image_url": Value("string"),
+                    "metadata_url": Value("string"),
+                    "original_height": Value("int32"),
+                    "original_width": Value("int32"),
+                    "mime_type": Value("string"),
+                    "caption_attribution_description": Value("string"),
+                    "wit_features": Sequence(
+                        {
+                            "language": Value("string"),
+                            "page_url": Value("string"),
+                            "attribution_passes_lang_id": Value("string"),
+                            "caption_alt_text_description": Value("string"),
+                            "caption_reference_description": Value("string"),
+                            "caption_title_and_reference_description": Value("string"),
+                            "context_page_description": Value("string"),
+                            "context_section_description": Value("string"),
+                            "hierarchical_section_title": Value("string"),
+                            "is_main_image": Value("string"),
+                            "page_changed_recently": Value("string"),
+                            "page_title": Value("string"),
+                            "section_title": Value("string"),
+                        }
+                    ),
+                }
+            ),
```

Comment:
Stupid question: API doesn't seem to correspond with https://github.com/google-research-datasets/wit/blob/main/DATA.md why is there a need to do so?

I would expect the `datasets` integration of WIT dataset to have the same API (maybe replacing `image_url` with downloaded image) but without significant changes.

Reply:
Hi! So the Google version is a processed version of the Wikimedia version, which has more information. I'd say the latter is more important to have, but the ideal would be to have two configs - one for this version for the script, and one for the one you wrote in https://github.com/huggingface/datasets/pull/4175

Continuing the conversation there